### PR TITLE
OT-3551 Backface culling in shaders

### PR DESCRIPTION
**Pull request by amilovsky**

### CODE REVIEW

File: keentools/utils/shaders.py [hunk 1]
```diff
@@ -445,35 +451,40 @@ def lit_fragment_shader() -> str:
...
-        light3.quadratic = 0.0;
-        light3.ambient = vec3(0.0, 0.0, 0.0);
-        light3.diffuse = vec3(1.0, 1.0, 1.0);
-
-        fragColor = vec4(
-            to_srgb_gamma_vec3(evaluatePointLight(light1, color.rgb, calcNormal, outPos)) +
-            to_srgb_gamma_vec3(evaluatePointLight(light2, color.rgb, calcNormal, outPos)) +
-            to_srgb_gamma_vec3(evaluatePointLight(light3, color.rgb, calcNormal, outPos)),
-            color.a * adaptiveOpacity);
+        if (ignoreBackface && (dot(calcNormal, camDir) < 0.0)) discard;
```
Comment (SergeyKrivohatskiy) [inline keentools/utils/shaders.py:454]: Refactor this. Fix formatting. Invert long if

File: keentools/geotracker/camera_input.py [hunk 2]
```diff
@@ -70,16 +70,28 @@ def image_size(self) -> Tuple[int, int]:
...
+    @classmethod
+    def _set_previous_val(cls, val) -> None:
         cls._previous_val = val
 
     def _rounded(self, val: float) -> int:
         p = frexp(val)
         return int(round(p[0] * 10000, 0) + 3 * p[1])
 
     def geo_hash(self) -> Any:
+        _log.output(_log.color('magenta', 'get geo_hash'))
```
Comment (SergeyKrivohatskiy) [inline keentools/geotracker/camera_input.py:94]: Remove this?

File: keentools/geotracker/camera_input.py [hunk 3]
```diff
@@ -88,15 +100,18 @@ def geo_hash(self) -> Any:
         geotracker = settings.get_current_geotracker_item()
         if geotracker and geotracker.geomobj:
             vert_count = len(geotracker.geomobj.data.vertices)
+            poly_count = len(geotracker.geomobj.data.polygons) * 3  # empirical
```
Comment (SergeyKrivohatskiy) [inline keentools/geotracker/camera_input.py:103]: Remove comment. Rename variable to `empirical_poly_count` =)

File: keentools/geotracker/camera_input.py (hunk 2 above)
Comment (SergeyKrivohatskiy) [reply keentools/geotracker/camera_input.py:94]: Here and in other places? Or this is hidden?

File: keentools/utils/gpu_shaders.py [hunk 4]
```diff
@@ -493,36 +498,41 @@ def lit_local_shader(use_old: bool=_use_old_shaders) -> Any:
...
-        light3.quadratic = 0.0;
-        light3.ambient = vec3(0.0, 0.0, 0.0);
-        light3.diffuse = vec3(1.0, 1.0, 1.0);
-
-        fragColor = vec4(
-            to_srgb_gamma_vec3(evaluatePointLight(light1, color.rgb, calcNormal, outPos)) +
-            to_srgb_gamma_vec3(evaluatePointLight(light2, color.rgb, calcNormal, outPos)) +
-            to_srgb_gamma_vec3(evaluatePointLight(light3, color.rgb, calcNormal, outPos)),
-            color.a * adaptiveOpacity);
+        if (ignoreBackface && (dot(calcNormal, camDir) < 0.0)) discard;
```
Comment (SergeyKrivohatskiy) [inline keentools/utils/gpu_shaders.py:501]: Refactor this

**Review by SergeyKrivohatskiy — CHANGES_REQUESTED** [review]

File: keentools/geotracker/camera_input.py (hunk 2 above)
Comment (amilovsky) [reply keentools/geotracker/camera_input.py:94]: This call is visible in debug mode only

**Review by SergeyKrivohatskiy — APPROVED** [review]

**Pull request merged**